libraries/AP_AHRS/AP_AHRS_DCM.cpp

```diff
@@ -1036,10 +1040,13 @@ bool AP_AHRS_DCM::get_position(struct Location &loc) const
     loc.terrain_alt = 0;
     loc.offset(_position_offset_north, _position_offset_east);
     const AP_GPS &_gps = AP::gps();
-    if (_flags.fly_forward && _have_position) {
-        float gps_delay_sec = 0;
-        _gps.get_lag(gps_delay_sec);
-        loc.offset_bearing(_gps.ground_course(), _gps.ground_speed() * gps_delay_sec);
+    if (_have_position) {
+        uint32_t now = AP_HAL::millis();
+        float dt = 0;
+        _gps.get_lag(dt);
+        dt += constrain_float((now - _last_pos_ms) * 0.001, 0, 0.5);
```

Comment:
Doing things this way has the advantage that every time you call it you get a more up-to-date position.

That's also a disadvantage in that other data that we calculate once per loop might become out-of-date compared to this position.  Not saying that's a blocker, but as I was trying to get the DCM results into a structure, I noted this sort behaviour as something that would always have to be passed through as a method call onto the active backend, rather than retrieved from the AHRS frontend copied-variables (e.g. AP::ahrs().roll is copied from the backend's data).  We couldn't take a copy of this projected location without losing the "most recent projection" property that's here now.

One option here is to NOT constrain the time here but to return `false` from the method if we're outside the constraints.  `_have_position` isn't directly related to `_last_pos_ms`.

Reply:
we could copy it - the projection is for between GPS samples (200ms) not between AHRS updates (at most 20ms, usually less)
but I want this PR for 4.1 to be as simple as possible, and I think this is simple and robust